sched/clock/clock_systime_timespec.c

```diff
@@ -62,21 +62,21 @@
 int clock_systime_timespec(FAR struct timespec *ts)
 {
 #ifdef CONFIG_RTC_HIRES
+  irqstate_t flags;
+
+  flags = spin_lock_irqsave(&g_basetime_lock);
   if (g_rtc_enabled)
     {
-      irqstate_t flags;
-
       up_rtc_gettime(ts);
```

Comment:
g_basetime_lock is just use for protect g_basetime

Reply:
add c_section protection to up_rtc_gettime in driver code

Reply:
Removing the global lock is to improve performance, but adding c_section will make the performance worse, right?

Reply:
This critical section is temporarily added. To avoid potential issues, it will be gradually removed entirely in the future. The goal is to replace the critical section with a spinlock.

Reply:
So what's the point of removing the global lock? Make the code more complicated and make the performance worse?We are not robots. If it is just a simple replacement of the lock, why not let AI do it? Any changes need to think about how to make the code better.

Reply:
Many cases are not simply a matter of replacement. 

There are roughly three purposes:
1 The current c_section is used too extensively, leading many meaningless busywaits. Using smaller locks can avoid competition among critical sections, thereby improving performance.
2 There are logical flaws in c_section, It doesn't match the apparent semantics. which can cause the protection provided by the c_section to fail in casees where wait/post is called.
for example
enter_critical_section
post/wait  ->>> protection failed 
leave_critical_section
3 The implementation of the c_section itself is complex and time-consuming. Using a spinlock is better than using a critical section.
4 Due to the auto locking and unlocking mechanism during context switching in c_section, we are unable to precisely control the duration of the lock, which leads to performance issues.

Reply:
I just only one opinion "We can't let the performance get worse after any change" right?

spin_lock_irqsave(NULL) ->
spin_lock_irqsave(lock) + enter_critical_section()

up_rtc_gettime() is called very frequently. If the timer driver needs enter_critical_section() protection, it means that the spinlock global lock is not really removed yet.

Reply:
The entire replacement process needs to be carried out step by step. We can quickly submit a patch to replace the critical sections in the up_rtc_gettime driver as well.

Reply:
Please
1. Resolve the lower half resource exclusive problem first
2. Then delete the global spinlock

Reply:
ok

Reply:
I added spinlock to protect up_rtc_gettime